ratis-common/src/main/java/org/apache/ratis/protocol/RaftPeer.java

```diff
@@ -163,9 +213,13 @@ public RaftPeerProto getRaftPeerProto() {
   @Override
   public String toString() {
     final String rpc = address != null? "|rpc:" + address: "";
+    final String admin = adminAddress != null && !Objects.equals(address, adminAddress)
+        ? "|admin:" + adminAddress : "";
+    final String client = clientAddress != null && !Objects.equals(address, clientAddress)
+        ? "|client:" + clientAddress : "";
```

Comment:
Nit: I think it is more readable to move "|admin:" and "|client:" before the `?` thus these string will always be generated.

Reply:
For brevity I'd rather print it only if it is different.